app/Http/Controllers/PagosController.php

```diff
@@ -146,10 +146,16 @@ public function savePettyCash(Request $request){
         $transaction->ref_payco = "1";
         $transaction->codigo_respuesta = "1";
         $transaction->respuesta = "Aprobado";
-        $transaction->amount = $request->amount;
         $transaction->data = $request->data ?? "";
         $transaction->user_id = $request->clientId;
         $transaction->created_at = $payDay;
+        if ($request->has('cxp') || $request->cxp == 1) {
```

Comment:
tiene que asegurar que el request SIEMPRE traiga el cxp

``
```suggestion
       $transaction->cxp = $request->cpx;
       $transaction->amount = $request->cpx ? -1*abs($request->amount) : $request->amount;
```